src/content/en/2021/javascript.md

```diff
@@ -1,20 +1,911 @@
+  sheets_gid="1511556407",
+  sql_file="lighthouse_unminified_js_bytes.sql"
+  )
+}}
+
+And it is found that 57.4% of mobile pages have 0 KB of unminified JS whereas 17.9% of those have 0-10 kilobytes of unminified JS.
+
+{# TODO I don't really understand what this graph is showing me. Especially compared to last graph. These sound small so is this good? Or bad? #}
+
+The first-party vs. third-party analysis in this case shows that 82% of the average mobile page's unminified JS bytes actually come from first-party scripts.
+
+{{ figure_markup(
+  image="average-unminified-js-bytes.png",
+  caption="Average distribution of unminified JS Bytes.",
+  description="Pie chart showing the average distribution of unminified JS bytes. 82% of the average mobile page's unminified JS bytes actually come from first-party scripts.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpHzC_cMZYj2VLzQ4ODK3uvZkNBXtwdAZriZaBwjLjUM1SGwwmJs9rv8T6OtNdXox29PQ34CasUUwc/pubchart?oid=1119550643&format=interactive",
+  sheets_gid="127709080",
+  sql_file="lighthouse_unminified_js_by_3p.sql"
+  )
+}}
+
+### Source maps
+
+Source maps are files sent along with the JavaScript resource files to let the browser map the minified or compressed file to their original version of that file. This is how source maps help us in debugging in the production environment.
+
+{{ figure_markup(
+  content="0.1%",
+  caption="Mobile pages using the sourcemap header.",
+  classes="big-number",
+  sheets_gid="1186092564",
+  sql_file="sourcemap_header.sql"
+  )
+}}
+
+It is found that only 0.1% of mobile pages do not use the SourceMap response header on script resources. One reason for this extremely small percentage could be that not many sites choose to put their original source code in production through the source map.
+
+An analysis of how many sites actually send the source map header on their first-party or third-party scripts shows:
+
+{{ figure_markup(
+  image="sourcemap-first-third-party.png",
+  caption="Usage of sourcemap headers in first party vs third party.",
+  description="Bar chart showing the usage of sourcemap headers in first party vs third party. 98% of the JS requests that include a SourceMap header on mobile are for first-party scripts and only 2% for third party",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpHzC_cMZYj2VLzQ4ODK3uvZkNBXtwdAZriZaBwjLjUM1SGwwmJs9rv8T6OtNdXox29PQ34CasUUwc/pubchart?oid=1762186682&format=interactive",
+  sheets_gid="2057978707",
+  sql_file="sourcemap_header_by_3p.sql"
+  )
+}}
+
+98% of the JS requests that include a SourceMap header on mobile are for first-party scripts.
+
+{# TODO That seems surprising to me. I would have though it was more common to be from third-parties since the likes of query and Bootstrap include source map files. Any thoughts on why this is the reverse? Or am I just wrong about this expectation? #}
```

Comment:
@NishuGoel

Reply:
added to the comment, please review

Reply:
As I'm editing the chapter I found the TODO and this discussion thread.

98% comes from the [pivot table](https://docs.google.com/spreadsheets/d/1zU9rHpI3nC6jTz3xgN6w13afW7x34xAKBh2IPH-lVxk/edit?disco=AAAAQ5pjLK8) that I've added to the sheet. Of the requests that have the header set, 230,623 are first-party and 4,745 are third-party, or 98% first-party.

Here's the [query](https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/sql/2021/javascript/sourcemap_header_by_3p.sql) for reference.

I don't think the way it was rewritten is correct:

> An analysis of how many sites actually send the source map header on their first-party or third-party scripts shows that 0.1% of the JavaScript requests that include a SourceMap header on mobile are for first-party scripts.

0.1 is not the % of JS requests with the header, but % of all JS requests.

Reply:
Sure @rviscomi 
We discussed it later. 
Could you revert it please to 98%?